control-plane/api-gateway/common/resources.go

```diff
@@ -429,6 +435,23 @@ func (s *ResourceMap) AddGatewayPolicy(gatewayPolicy *v1alpha1.GatewayPolicy) *v
 	return s.gatewayPolicies[key]
 }
 
+func (s *ResourceMap) AddJWTProvider(provider *v1alpha1.JWTProvider) {
+	key := api.ResourceReference{
+		Kind: provider.Kind,
+		Name: provider.Name,
+	}
+	s.jwtProviders[key] = provider
+}
+
+func (s *ResourceMap) GetJWTProviderForProvider(provider *v1alpha1.GatewayJWTProvider) (*v1alpha1.JWTProvider, bool) {
```

Comment:
The name of this function causes me to rethink what's happening every time it's used. I kind of wish our `v1alpha1.GatewayJWTProvider` was `...JWTProviderReference` and this function was `GetJWTProviderForReference`. Is this just me?

Reply:
yeah, I've been hitting more and more weirdness with the name, I'm down with changing the name, though `JWTProviderReference` isn't as descriptive because this also contains the claim information about what we should be verifying

Reply:
That makes sense. Maybe even `GetJWTProviderForGatewayJWTProvider` at the risk of being too verbose. It's not a blocker, just a stumbling point when reading code

Reply:
I can do that, also down to have more of a brainstorming session to see if we can name these a bit better